microsoft-edge/webview2/howto/chromium-devtools-protocol.md

```diff
@@ -0,0 +1,190 @@
+
+1.  Open Visual Studio.  
+1.  Choose **Project** > **Manage NuGet Packages** > **Browse** > **Include pre-release**.
+1.  Type `Microsoft.Web.WebView2.DevToolsProtocolExtension` and choose **Microsoft.Web.WebView2.DevToolsProtocolExtension** > **Install**.  
+    
+> [!NOTE] 
+> Ensure you also have the WebView2 NuGet Package named `Microsoft.Web.WebView2` installed.  
+
+:::image type="complex" source="./media/cdpnuget.png" alt-text="Ensure Microsoft.Web.WebView2.DevToolsProtocolExtension displays in the Visual Studio NuGet Package Manager" lightbox="./media/cdpnuget.png":::
+   Ensure **Microsoft.Web.WebView2.DevToolsProtocolExtension** displays in the Visual Studio NuGet Package Manager  
+:::image-end:::  
+
+## Step 4: Use DevTools Protocol Helper  
+
+1.  Add the `DevToolsProtocolExtension` and `CoreWebView2` namespaces to your project.
+    
+    ```csharp
+    using Microsoft.Web.WebView2.Core;
+    using Microsoft.Web.WebView2.Core.DevToolsProtocolExtension;
+    ```  
+    
+1.  Run the `GetDevToolsProtocolHelper()` method to instantiate the `DevToolsProtocolHelper` object, and navigate to `geolocater.html`.
+    
+    ```csharp
+    async void InitializeAsync()
+    {
+        await webView.EnsureCoreWebView2Async(null);
+        DevToolsProtocolHelper helper = webView.CoreWebView2.GetDevToolsProtocolHelper(); 
+        
+        webView.CoreWebView2.Navigate(@"C:\{PATH TO FILE}\geolocater.html");
+    }
+    ```  
+    
+1.  Run the [setGeoLocationOverrideAsync][setGeoLocationOverrideAsync] method.  For more information, navigate to [method-setGeolocationOverride][GithubChromedevtoolsDevtoolsProtocolTotEmulationMethodSetgeolocationOverride].  
+    
+    ```csharp
+    async void InitializeAsync()
+    {
+        await webView.EnsureCoreWebView2Async(null);
+        DevToolsProtocolHelper helper = webview.CoreWebView2.GetDevToolsProtocolHelper();
+        
+        webView.CoreWebView2.Navigate(@"C:\{PATH TO FILE}\geolocater.html");
+        
+        // Latitude and longitude for Paris, France.
+        double latitude = 48.857024082572565;  
+        double longitude = 2.3161581601457386;  
+        double accuracy = 1;
+        await helper.Emulation.setGeolocationOverrideAsync(latitude, longitude, accuracy);
+        
+    }
+    ```  
+    
+1.  Run your app.  
+1.  To display the coordinates of Paris, France, choose the **Display Location** button.  
+    
+    :::image type="complex" source="./media/finallocation-cdp.png" alt-text="The HTML file is displayed in a WebView2 control with the coordinates for Paris" lightbox="./media/finallocation-cdp.png":::
+       The HTML file is displayed in a WebView2 control with the coordinates for Paris  
+    :::image-end:::
+    
+## File a Chromium DevTools Protocol bug  
+
+To file a Chromium DevTools Protocol bug or issue, complete the following actions.  
+
+1.  Navigate to [WebView Feedback][GithubMicrosoftedgeWebview2feedback].  
```

Comment:
Did you talk with folks about this? We don't own the implementations for those huge number of CDP method/events. Are we planning on having bugs in our DB and then opening them against chromium for the end dev?